File: libraryFinal/src/main/java/com/library/Repository/ListDb.java
package com.library.Repository;

import com.library.Model.*;
import com.library.Controller.*;

import java.util.ArrayList;
import java.util.List;

public class ListDb {
    
    List<Product> listArray;

    public ListDb(){
        this.listArray = new ArrayList<Product>();
    }

    public void getListArray() {
        listArray.forEach(listArray -> {
        System.out.println("Percorrendo: ");
        System.out.println("List" + listArray);
        });
    }

    public void setListArray(Product prod) {
        this.listArray.add(prod);
    }
}

File: libraryCSharp/Repository/ListDb.cs
using System;
using System.Collections.Generic;
using libraryCSharp.Model;
using System.Collections;

namespace libraryCSharp.Repository
{
    public class ListDb
    {
        private List<Product> list;
        
        public ListDb(){
            this.list= new List<Product>();
        }
        
        public void getListArray() {
            foreach(Product product in list){
                Console.WriteLine(product.ToString());
            }
        }

        public void setList(Product prod)
        {
            this.list.Add(prod);
        }
    }
}